plugin/src/main/groovy/com/novoda/staticanalysis/internal/CodeQualityConfigurator.groovy

```diff
@@ -27,16 +30,19 @@ abstract class CodeQualityConfigurator<T extends SourceTask, E extends CodeQuali
             project.extensions.findByType(extensionClass).with {
                 defaultConfiguration.execute(it)
                 ext.exclude = { Object rule -> sourceFilter.exclude(rule) }
+                ext.includeVariants = { Closure<Boolean> filter -> includeVariantsFilter = filter }
                 config.delegate = it
                 config()
             }
             project.afterEvaluate {
                 boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
                 boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
                 if (isAndroidApp || isAndroidLib) {
-                    configureAndroidProject(isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants)
-                    configureAndroidProject(project.android.testVariants)
-                    configureAndroidProject(project.android.unitTestVariants)
+                    NamedDomainObjectSet<Object> variants = project.container(Object)
```

Comment:
I actually tried, but it was failing to execute `addAll()` with a cryptic stacktrace that sounded like a covariance issue.
The rationale of using `NamedDomainObjectSet` is that every `project.android.*Variants` is already of this type, and it supports an elegant `matching()` that reads better than the `findAll()` provided in `Collection`.